salt/elasticsearch/init.sls

```diff
@@ -0,0 +1,34 @@
+elasticsearch-deps:
+    pkg.installed:
+        - name: apt-transport-https
+
+elasticsearch:
+    # add repo to managed repositories
+    pkgrepo.managed:
+        - name: deb https://artifacts.elastic.co/packages/7.x/apt stable main
+        - key_url: salt://elasticsearch/release-key
+        - require:
+            - pkg: elasticsearch-deps
+    # install elastic search package
+    # TODO versioning
```

Comment:
If we don't need anything more specific than 7.x atm this is probably fine and we can add more specifics later, so you can remove the TODO.